### ISSUE TITLE
disconnections

### ISSUE THREAD

**Issue by danielskatz** [Parsl/parsl#18]
What happens if a Parsl code starts an app, then disconnects?

For example, on an HPC system, Parsl is running on a front end node and launches a long-running HPC app, with another app that depends on it.  While the HPC app is running, the front end node is restarted.  The HPC app keeps running.  Will the dependent app ever run?

**Comment by yadudoc** [comment]
Right now, all workflow state is held in the python interpreter (running on the headnode). If the headnode reboots, all state is lost and the workers executing the apps will abandon their tasks once they realize that the controller is lost. 

For long running workflows, it might make sense to use a reliable store for all state such that workflow state can be recovered. Right now, the dfk holds the state of dependent apps and decides that an app's dependencies have resolved and is ready to run. Some of this logic could be pushed to this "reliable store".

Does that answer your question ?

**Comment by danielskatz** [comment]
yes - I mostly wanted to raise this issue so that we could discuss possible additions to the roadmap relative to it.

**Comment by danielskatz** [comment]
I would label this as enhancement if I had permission to add labels...

**Comment by annawoodard** [comment]
I think this is addressed now with checkpointing. @yadudoc @danielskatz @kylechard what do you think?

**Comment by yadudoc** [comment]
We address the apps that have completed via checkpointing. Tasks that have been farmed out to workers will fail, once the workers are unable to call back to the parsl dfk process and timeout. The rationale here is that if the client has disconnected we have no way of knowing when the workers could connect back, and we don't want to waste resources waiting.

**Comment by annawoodard** [comment]
Right. So is any further action needed on this?
